### PR TITLE
add connection open properties

### PR DESCRIPTION
**Pull request by devigned**
Resolves: #28 

- add properties field to the conn struct
- add ConnOption to configure properties
- set properties on the open performative
- fix integration tests to target the exact API version of the Azure Service Bus management lib to ensure no future breaking changes

### CODE REVIEW

File: conn_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	c := new(conn)
+	key := "propKey"
+	value := "propValue"
+	err := ConnProperty(key, value)(c)
+	if err != nil {
+		t.Error(fmt.Sprintf("%+v", err))
+	}
+
+	if c.properties[symbol(key)] != value {
+		t.Errorf("conn property was not set to key \"%s\" and value \"%s\"", key, value)
```
Comment (vcabbage) [inline conn_test.go]: Just FYI, the `%q` formatter is handy when you want to quote strings. It's particularly useful when comparing strings that may have unprintable characters. https://play.golang.org/p/xx2QI3_4Irr

File: conn_test.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"testing"
+)
+
+func TestConnProperty(t *testing.T) {
+	c := new(conn)
+	key := "propKey"
+	value := "propValue"
+	err := ConnProperty(key, value)(c)
+	if err != nil {
+		t.Error(fmt.Sprintf("%+v", err))
```
Comment (vcabbage) [inline conn_test.go]: I don't know how this sort of thing ended up in the other tests, but it should just be `t.Errorf("%+v", err)`. I need to correct my usages.

File: conn_test.go (hunk 1 above)
Comment (devigned) [reply conn_test.go]: Oh, that's fancy. Thanks for spreading the knowledge.

**Comment by devigned** [comment]
@vcabbage I added an error check for empty keys. I don't think an empty symbol makes sense.

**Comment by vcabbage** [comment]
Thanks!

**Comment by devigned** [comment]
My pleasure. Thank you for the feedback and the quick merge.

**Pull request merged**